### PR TITLE
refactor(core): consolidate backend raw responses onto mot.raw namespace

### LINKED ISSUE #1215: refactor(core): consolidate backend raw responses onto mot.raw namespace
**Sub-issue of #909.** Design discussed in [#1197 Area 1](https://github.com/generative-computing/mellea/discussions/1197).

## Summary

Each backend currently writes a different bag of provider-specific keys to `mot._meta` (`chat_response`, `oai_chat_response`, `litellm_chat_response`, `hf_output`, etc.). This issue replaces those writes with a typed `mot.raw: RawProviderResponse` namespace.

## Motivation

`mot._meta` is `CBlock` surface, but every MOT also stuffs backend-native response objects into it. Consumers reading these values (e.g. `chat.py:_parse`) have no signal that they're opting into provider-coupled access. A typed sibling of `mot.generation` — `mot.raw` — makes the escape hatch explicit while leaving `mot._meta` for `CBlock`-level metadata.

## Design

```python
@dataclass
class RawProviderResponse:
    """Backend-native response payload. Provider-coupling escape hatch.

    Anyone reading these fields is opting into provider-specific shape. Portable
    consumers should read mot.value, mot.parsed_repr, mot.tool_calls, or
    mot.generation instead.
    """
    provider: str | None = None
    response: Any | None = None              # full SDK response object
    streamed_chunks: list[Any] | None = None # streaming-only
```

- `mot.raw.response` replaces `chat_response` / `oai_chat_response` / `litellm_chat_response` / `hf_output`.
- `mot.raw.streamed_chunks` replaces the parallel `*_streamed` lists.
- `*_streaming_usage` keys disappear — already covered by `mot.generation.usage`.
- `oai_chat_response_choice` is dropped — always equals `oai_chat_response["choices"][0]`.

`response` is typed `Any`. The whole point of `raw` is opt-in provider coupling; portable structured access is on `mot.generation`.

## `chat.py:_parse` migration

Switches on `mot.raw.provider`:
- `"ollama"` → `response.message.role` / `response.message.content` (role can be `"assistant"` or `"tool"`)
- `"openai"` / `"watsonx"` → `response["choices"][0]["message"]…`
- HuggingFace fallback unchanged

The role-recovery branch stays — Ollama can legitimately return `role="tool"` ([SDK docstring](https://github.com/ollama/ollama-python)), so hardcoding `"assistant"` would silently change behavior for tool responses. ([OQ2 resolution](https://github.com/generative-computing/mellea/discussions/1197))

## Migration

Single PR:
1. Backends migrate writes from `_meta[<key>]` to `mot.raw.response` / `mot.raw.streamed_chunks`.
2. Internal readers migrate (`chat.py:_parse`, tests).
3. `_meta` becomes a read-only deprecation shim emitting `DeprecationWarning` for the old keys, for one minor release.

After one minor cycle: shim removed.

## Appendix items folded in

From #1197's Appendix A, all to verify or address during this migration:

- **Item 3** — `chat.py:_parse` HuggingFace fallback hardcodes `role="assistant"` and returns raw text without role extraction. Confirm during migration whether this fallback is still desired with `mot.raw.response` available, or whether a richer shape can be extracted.
- **Item 4** — `oai_chat_response_choice` is redundant with `oai_chat_response["choices"][0]`. Drop in this PR (both `openai.py` and `watsonx.py` write it).
- **Item 5** — `chat.py:_parse` HuggingFace tool-call branch returns `Message(role="assistant", content=computed.value)` without recovering tool-call structure. Decide whether to serialize `computed.tool_calls` instead during migration.

## Out of scope

- **Logits (#123)**: per #793 precedent, logits are execution metadata and belong on `mot.generation` (e.g. `generation.logprobs`), not `mot.raw`.
- **Per-provider Union typing on `response`**: `Any` is correct ([OQ1 resolution](https://github.com/generative-computing/mellea/discussions/1197)).
- **Migrating raw-path `_meta["usage"]` to `mot.generation.usage`**: separate bug fix (#1218), not #909 scope.

## Acceptance

- [ ] `mot.raw` populated by all five backends (ollama, openai, watsonx, litellm, huggingface) on chat path
- [ ] `mot.raw.streamed_chunks` populated on streaming path
- [ ] `chat.py:_parse` switches on `mot.raw.provider`
- [ ] `_meta[<old key>]` reads emit `DeprecationWarning`
- [ ] `oai_chat_response_choice` removed
- [ ] Tests updated; existing behavior preserved

## Sequencing

Independent of #1181. Can ship after #1181 merges (which closes #909 Area 2) but before or after Area 3 (#1216) / Area 4 (#1217).

### PR DESCRIPTION
**Pull request by ajbozarth**
# Pull Request

## Issue
Fixes #1215

## Description

Replaces the per-backend bag of provider-coupled `mot._meta` keys with a typed `mot.raw: RawProviderResponse` namespace — a sibling to `mot.generation`.

Each backend used to stuff its native SDK response into `mot._meta` under a different key (`chat_response`, `oai_chat_response`, `oai_chat_response_choice`, `litellm_chat_response`, `litellm_full_response`, `oai_completion_response`, `generate_response`, `hf_output`, plus `*_streamed` / `*_streaming_usage` siblings). `_meta` is `CBlock`-level metadata, so consumers reading these values (e.g. `chat.py:_parse`) had no signal they were opting into provider-coupled access. A typed `mot.raw` makes that escape hatch explicit and leaves `_meta` for `CBlock`-level metadata.

### What changed

- **New type** `RawProviderResponse(provider, response, streamed_chunks)` in `mellea/core/base.py`, exported from `mellea.core`. Carried through `ModelOutputThunk.__init__`, `__copy__` (shallow), `__deepcopy__` (deep), and `_copy_from` (reference), matching `mot.generation` semantics.
- **All five backends migrated** (ollama, openai, watsonx, litellm, huggingface) on both the chat path (`processing` / `post_processing`) and the raw path (`_generate_from_raw`). Each sets `mot.raw.response` (and `mot.raw.streamed_chunks` for streaming) plus `mot.raw.provider`.
- **`chat.py:_parse` dispatches on `mot.raw.provider`** instead of probing `_meta` keys. The Ollama role-recovery branch is preserved (Ollama can return `role="tool"`).
- **`cli/serve/utils.py:extract_finish_reason`** also migrated to switch on `mot.raw.provider`.
- **`oai_chat_response_choice` removed.** It was redundant with the choice already inside the response; tool extraction now reads the response directly per its per-path shape.
- **Streaming usage** (`oai_streaming_usage`, `litellm_streaming_usage`) is no longer stashed in `_meta`; it is written straight to `mot.generation.usage` when the usage chunk arrives.

### Reviewer callouts

- **No deprecation shim.** The issue body called for `_meta` to become a read-only deprecation shim. We chose **not** to add one: the `_meta` provider keys are private (underscored) and undocumented, and the project's convention is to run a deprecation cycle only for public surface. Reading a removed key now returns `None` rather than warning.
- **Per-path response shape is preserved, not unified.** Non-streaming stores the full top-level response dict; streaming stores the already-merged choice-level dict (what `chat_completion_delta_merge` returns). Every consumer that reads `mot.raw.response` for OpenAI-shaped backends detects which shape it has (`"choices" in response`): the backends' `post_processing` tool extraction, `cli/serve/utils.py:extract_finish_reason`, and `chat.py:_parse` (both the tool and non-tool branches). Unifying these to a single shape is a real cleanup but a behavior change, so it is deferred (see https://github.com/generative-computing/mellea/issues/1330).
- **`mot.raw.response` is always `dict | None` for the OpenAI-shaped backends.** All three (openai, litellm, watsonx) store a dict on both paths (`chunk.model_dump()` / the merged-delta dict / the WatsonX `chunk` dict), so `post_processing` reads `usage` via `assert response is not None` + `response.get("usage")` without a non-dict fallback. The old `getattr(response, "usage", None)` guard protected a non-dict case that cannot occur on these paths.
- **WatsonX wrote the `oai_*` keys**, not a `watsonx_*` namespace. The migration neutralizes that drift — all backends now use `mot.raw`.
- **`litellm_full_response` was undocumented** in the issue body but existed in code. It and `litellm_chat_response` are folded into a single `mot.raw.response` holding the top-level shape.
- **HuggingFace `_parse` fallback retained.** HF's `mot.raw.response` is a `GenerateDecoderOnlyOutput` (token tensors) with no role/content surface beyond `mot.value`, so the `Message(role="assistant", content=mot.value)` fallback stays unchanged from before.
- **`mot.generation.provider` and `mot.raw.provider` duplication** is intentional: a consumer reading `mot.raw` shouldn't have to also reach into `mot.generation` to learn the provider that determines `response`'s shape.

### Testing
- [x] Tests added to the respective file if code was changed
- [x] New code has 100% coverage if code was added
- [ ] Ensure existing tests and github automation passes (a maintainer will kick off the github automation when the rest of the PR is populated)

Local verification:
- `uv run pytest test/core/test_base.py test/backends/test_ollama_unit.py test/backends/test_openai_unit.py test/backends/test_litellm_thinking.py test/backends/test_huggingface_unit.py test/stdlib/components/test_chat.py test/cli/test_serve_utils.py -m "not qualitative"` — all pass.
- `ruff format --check .`, `ruff check .`, `mypy .` — clean.
- Docstring quality gate (`audit_coverage.py --quality --fail-on-quality --threshold 100`) — 100% coverage, 0 issues (`RawProviderResponse` is new in `__all__`).
- `grep -rn '_meta\[' mellea/` — no provider keys remain (only `CBlock` constructor/repr).

### Attribution
- [x] AI coding assistants used

---

### Adding a new component, requirement, sampling strategy, or tool?

- [ ] Component
- [ ] Requirement
- [ ] Sampling Strategy
- [ ] Tool

### CODE REVIEW

**Comment by ajbozarth** [comment]
I opened https://github.com/generative-computing/mellea/issues/1330 and https://github.com/generative-computing/mellea/issues/1331 as follow up issues. 

https://github.com/generative-computing/mellea/issues/1330 in particular could be argued that it belongs in this PR, but I left it out in my first pass. @jakelorocco if you think I should include that refactor here I'll add it, or if you think that behavior is "as intended" I can close it.

Similarly if https://github.com/generative-computing/mellea/issues/1331 is "behaves as intended" I can close it. I wanted to include that fix in my work here like I did for usage in my metrics work, but Claude specifically called out the kv_cache as non-trivial to cleanly slice into MOTs

**Comment by jakelorocco** [comment]
> I opened #1330 and #1331 as follow up issues.
> 
> #1330 in particular could be argued that it belongs in this PR, but I left it out in my first pass. @jakelorocco if you think I should include that refactor here I'll add it, or if you think that behavior is "as intended" I can close it.
> 
> Similarly if #1331 is "behaves as intended" I can close it. I wanted to include that fix in my work here like I did for usage in my metrics work, but Claude specifically called out the kv_cache as non-trivial to cleanly slice into MOTs

These both seem reasonable to defer. I marked #1331 as low priority because I don't think the raw endpoints are getting a lot of use right now.

**Review by jakelorocco — APPROVED** [review]
this seems like a reasonable change and continues to bring much needed structure to mots; thank you
